Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -269,6 +325,42 @@ open class NavigationMapView: UIView {
         applyRoutesDisplay(layerPosition: layerPosition)
     }
     
+    /**
+     Visualizes the given routes and it's alternatives, removing any existing from the map.
+     
+     Each route is visualized as a line. Each line is color-coded by traffic congestion, if congestion
+     levels are present. Implement `NavigationMapViewDelegate` methods to customize the appearance of
+     the lines representing the routes. To also visualize waypoints and zoom the map to fit,
+     use the `showcase(_:animated:)` method.
+     
+     To undo the effects of this method, use `removeRoutes()` and `removeContinuousAlternativesRoutes()` methods.
+     
+     - parameter routeResponse: `IndexedRouteResponse` containing routes to visualize. The selected route by `routeIndex` is considered primary, while the remaining routes are displayed as if they are currently deselected or inactive.
+     - parameter layerPosition: Position of the first route layer. Remaining routes and their casings
+     are always displayed below the first and all other subsequent route layers. Defaults to `nil`.
+     If layer position is set to `nil`, the route layer appears below the bottommost symbol layer.
+     - parameter legIndex: The zero-based index of the currently active leg along the primary route.
+     The active leg is highlighted more prominently than inactive legs.
+     */
+    public func show(_ routeResponse: IndexedRouteResponse,
+                     layerPosition: MapboxMaps.LayerPosition? = nil,
+                     legIndex: Int? = nil) {
+        guard let mainRoute = routeResponse.currentRoute else {
+            return
+        }
+        
+        removeRoutes()
+        removeContinuousAlternativesRoutesLayers()
+        
+        self.routes = [mainRoute]
+        self.continuousAlternatives = routeResponse.parseAlternativeRoutes()
+        currentLegIndex = legIndex
+        customRouteLineLayerPosition = layerPosition
+        
+        applyRoutesDisplay(layerPosition: layerPosition)
+        updateRouteLineWithRouteLineTracksTraversal()
```

Comment:
Is `updateRouteLineWithRouteLineTracksTraversal` call necessary here? According to what I see in the implementation details, this will cause the call of `show(_ routes:layerPosition:legIndex:)` and the second call of `removeRoutes`, `removeContinuousAlternativesRoutesLayers`, `applyRoutesDisplay(layerPosition:)`

Reply:
You are right, that's excessive.